forc/src/ops/forc_explorer.rs

```diff
@@ -0,0 +1,149 @@
+
+#[derive(Deserialize, Debug)]
+struct GitHubRelease {
+    url: String,
+    assets: Vec<GitHubReleaseAsset>,
+}
+
+#[derive(Deserialize, Debug)]
+struct GitHubReleaseAsset {
+    browser_download_url: String,
+}
+
+const REPO_RELEASES_URL: &str = "https://api.github.com/repos/FuelLabs/block-explorer-v2/releases";
+
+struct EndPoints {}
+
+impl EndPoints {
+    pub fn static_files() -> String {
+        "static".to_string()
+    }
+}
+
+struct ExplorerAppPaths {}
+
+impl ExplorerAppPaths {
+    pub fn web_app_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer")
+    }
+    pub fn build_archive_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/build.tar")
+    }
+    pub fn build_archive_unpack_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/build")
+    }
+    pub fn web_app_files_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/www")
+    }
+    pub fn web_app_static_assets_path() -> PathBuf {
+        dirs::home_dir().unwrap().join(".fuel/explorer/www/static")
+    }
+}
+
+pub(crate) async fn exec(command: ExplorerCommand) -> Result<(), reqwest::Error> {
+    let ExplorerCommand { port } = command;
+    if !has_static_files() {
+        let download_url = match get_release_url().await {
+            Ok(url) => url,
+            Err(error) => panic!("Failed to get release {:?}", error),
+        };
+        eprintln!("Downloading Fuel Explorer ...");
+
+        match download_build(&download_url).await {
+            Ok(arch) => arch,
+            Err(error) => panic!("Failed to download build {:?}", error),
+        };
+
+        match unpack_archive() {
+            Ok(_) => (),
+            Err(error) => panic!("Failed to unpack build archive {:?}", error),
+        };
+
+        if let Err(error) = rename(
+            ExplorerAppPaths::build_archive_unpack_path(),
+            ExplorerAppPaths::web_app_files_path(),
+        ) {
+            panic!("Failed to move static files {:?}", error)
+        }
+
+        match remove_file(ExplorerAppPaths::build_archive_path()) {
+            Ok(_) => (),
+            Err(error) => eprintln!("Failed clean up files {:?}", error),
+        }
+    }
+
+    start_server(port).await;
+    Ok(())
+}
+
+fn has_static_files() -> bool {
+    ExplorerAppPaths::web_app_files_path()
+        .join("index.html")
+        .exists()
+}
+
+async fn get_release_url() -> Result<String, reqwest::Error> {
+    let client = reqwest::Client::new();
+    let releases_url = REPO_RELEASES_URL;
+    let response = client
+        .get(releases_url)
+        .header("User-Agent", "warp")
+        .send()
+        .await?;
+    let response_json: Vec<GitHubRelease> = response.json().await?;
+    let download_url = response_json
+        .first()
+        .unwrap()
+        .assets
+        .first()
+        .unwrap()
+        .browser_download_url
+        .clone();
```

Comment:
That's a mouthful 😅. Do you mind adding a comment explaining what we're looking for inside the `response_json`? It might not be clear to someone without full context trying to debug/extend it.

Reply:
I refactored this logic, so comments not needed anymore